smsdk/_version.py

```diff
@@ -0,0 +1,113 @@
+#
+"""
+Track version information for sightmachine-sdk module here.
+"""
+import typing
+import itertools
+import requests
+import warnings
+import functools
+import re
+
+
+class VersionInfo(typing.NamedTuple):
+    major: typing.Any
+    minor: typing.Any
+    patchlevel: typing.Any
+    releaselevel: typing.Any
+    serial: typing.Any
+
+    def __str__(self) -> str:
+        result = "".join(
+            itertools.chain(
+                ".".join(str(i) for i in self[:3] if i is not None),
+                (str(i) for i in self[3:] if i is not None),
+            )
+        )
+        return result
+
+
+version_info = VersionInfo(1, 26, 0, "", None)
+version = str(f"v{version_info}")
+
+
+def sort_releases_descending(
+    releases: typing.List[typing.Dict[str, typing.Any]]
+) -> typing.List[typing.Dict[str, typing.Any]]:
+    def version_key(
+        release: typing.Dict[str, typing.Any]
+    ) -> typing.Union[typing.Tuple[int, int, int, str], typing.Tuple[()]]:
+        version = release["tag_name"]
+        # Use regular expression to extract numerical and string parts
+        match = re.match(r"(?i)v(\d+)\.(\d+)\.(\d+)(\D*)", version)
+        if match:
+            major, minor, patch, suffix = map(match.group, (1, 2, 3, 4))
+            return int(major), int(minor), int(patch), suffix
```

Comment:
FYI, I have seen an interesting way of splitting items for natural sorts.
```python
>>> [i for i in re.split(r'(\d+|\W+)', '3.40.5+ok3'.lower()) if i]
['3', '.', '40', '.', '5', '+', 'ok', '3']
>>> [i for i in re.split(r'\W+', '3.40.5+ok3'.lower()) if i]
['3', '40', '5', 'ok3']


def version_key(version:str) -> t_.Sequence:
    return  [
        (int(i) if i.isdigit() else i)
        for i in re.split(r'(\d+|\W+)', '3.40.5+ok3'.lower())
        if i
    ]
```

Reply:
Adopted this new function for sorting the releases aray.